ms-windows/build-tools.php

```diff
@@ -0,0 +1,1564 @@
+
+
+			// Nothing is perfect.  Patch the makefile.
+			$data = file_get_contents($pathmap["[[SOURCE_TEMP_DIR]]"] . "/makefile");
+
+			foreach ($archinfo["build"]["patches"] as $key => $val)
+			{
+				$data = str_replace($key, $val, $data);
+			}
+
+			// The 'no-shared' build should only generate statically linked libraries and executables.
+			if (strpos($configuretype, "no-shared") !== false)
+			{
+				$data = str_replace("MODULES=providers\\legacy.dll", "MODULES=", $data);
+				$data = str_replace("MODULEPDBS=providers\\legacy.pdb", "MODULEPDBS=", $data);
+			}
+
+			// Fix the build type.
+			$data = str_replace(" /MTd ", " " . $buildtype . " ", $data);
+			$data = str_replace(" /MTd\r\n", " " . $buildtype . "\r\n", $data);
+			$data = str_replace(" /MT ", " " . $buildtype . " ", $data);
+			$data = str_replace(" /MT\r\n", " " . $buildtype . "\r\n", $data);
+
+			$data = str_replace(" /MDd ", " " . $buildtype . " ", $data);
+			$data = str_replace(" /MDd\r\n", " " . $buildtype . "\r\n", $data);
+			$data = str_replace(" /MD ", " " . $buildtype . " ", $data);
+			$data = str_replace(" /MD\r\n", " " . $buildtype . "\r\n", $data);
+
+			file_put_contents($pathmap["[[SOURCE_TEMP_DIR]]"] . "/makefile", $data);
+
+
+			// Run nmake.
+			$cmd = escapeshellarg("nmake.exe") . " -f makefile";
+			echo $echoprefix . " Running '" . $cmd . "'...\n";
+
+			$errorfilename = $pathmap["[[BUILD_OUTPUT_DIR]]"] . "/logs/" . $buildnum . "_nmake_" . $outdir . "_error.log";
+			$result = RunCommand($cmd, $errorfilename, $pathmap["[[SOURCE_TEMP_DIR]]"], $baseenv);
+			if ($result === false)  CLI::DisplayError($echoprefix . " Failed to start process.");
+
+			file_put_contents($pathmap["[[BUILD_OUTPUT_DIR]]"] . "/logs/" . $buildnum . "_nmake_" . $outdir . "_out.log", $result);
+
+			if (!file_exists($pathmap["[[SOURCE_TEMP_DIR]]"] . "/apps/openssl.exe"))  CLI::DisplayError($echoprefix . " Failed to generate 'apps/openssl.exe'.");
+
+
+			// Copy files to the build output directory.
+			$indir = $pathmap["[[SOURCE_TEMP_DIR]]"];
+			$outdir2 = $pathmap["[[BUILD_OUTPUT_DIR]]"] . "/" . $outdir;
+			@mkdir($outdir2, 0775, true);
+			if (in_array("all", $copymodes) || in_array("apps", $copymodes))  CopyBuildFiles($indir, $outdir2, "apps", true);
+			if (in_array("all", $copymodes) || in_array("engines", $copymodes))  CopyBuildFiles($indir, $outdir2, "engines", false);
+			if (in_array("all", $copymodes) || in_array("fuzz", $copymodes))  CopyBuildFiles($indir, $outdir2, "fuzz", false);
+			if (in_array("all", $copymodes) || in_array("providers", $copymodes))  CopyBuildFiles($indir, $outdir2, "providers", false);
+			if (in_array("all", $copymodes) || in_array("test", $copymodes))  CopyBuildFiles($indir, $outdir2, "test", true);
+			if (in_array("all", $copymodes) || in_array("tools", $copymodes))  CopyBuildFiles($indir, $outdir2, "tools", true);
```

Comment:
So this does not use make install at all... Not sure if this is a problem or not.